### PR TITLE
Support removing an item from the cart

### PR DESCRIPTION
**Pull request by Deepak-Testpress**

### CODE REVIEW

File: myshop/tests/cart/test_views.py [hunk 1]
```diff
@@ -18,21 +18,37 @@ def setUp(self):
...
+        cart_detail_url = reverse("cart:cart_detail")
+        response = self.client.post(cart_remove_url)
+
+        self.assertEqual(cart_detail_url, response.url)
 
     def test_templates_used_for_cart_detail_view(self):
-        self.cart_detail_url = reverse("cart:cart_detail")
-        response = self.client.get(self.cart_detail_url)
+        cart_detail_url = reverse("cart:cart_detail")
+        response = self.client.get(cart_detail_url)
```
Comment (waranlogesh) [inline myshop/tests/cart/test_views.py:36]: Shouldn't be part of this PR. You can change in the PR when this code is added if the PR is still not merged or you can directly push this change to master and rebase this branch

File: myshop/cart/views.py [hunk 2]
```diff
@@ -25,3 +25,11 @@ def cart_add(request, product_id):
 def cart_detail(request):
     cart = Cart(request)
     return render(request, "cart/detail.html", {"cart": cart})
+
+
+@require_POST
+def cart_remove(request, product_id):
+    cart = Cart(request)
+    product = get_object_or_404(Product, id=product_id)
+    cart.remove_product(product)
```
Comment (waranlogesh) [inline myshop/cart/views.py:34]: Test case for remove_product() alone

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: myshop/cart/views.py (hunk 2 above)
Comment (Deepak-Testpress) [reply myshop/cart/views.py:34]: Test case for `remove_product` method has already been written in [test_cart.py](https://github.com/Deepak-Testpress/shopping_app/blob/98be6ff82b792c44531a2231e7acb4d5f5532b1e/myshop/tests/cart/test_cart.py#L104)

File: myshop/tests/cart/test_views.py (hunk 1 above)
Comment (Deepak-Testpress) [reply myshop/tests/cart/test_views.py]: Resolved by refactoring the code in PR where the code was added

**Review by waranlogesh — APPROVED** [review]
